### ISSUE TITLE
Dual wielding

### ISSUE THREAD

**Issue by TypicalDarkness** [Fexcraft/FlansMod#15]
I found a pretty reliable (but also hacky) way to prevent the player from equipping guns that don't have the oneHanded property set to true to the offHand, but I still couldn't decide on how the guns that CAN be dual wielded should behave. 
In which cases should left-clicking cancel the vanilla action and cause a shot instead?
- Never cancel vanilla, only shoot if there is also a gun in the mainHand (If so, does that gun have to be oneHanded aswell?):

- Only shoot if there is a gun in the mainHand or the mainHand is empty.

- Always cancel vanilla action and shoot, no matter what's in the mainHand.

(I have already posted this question on the 'Mod is cancelled forever?' issue on FlansMods/FlansMod, but I wanted to give some more explanation)

**Comment by Fexcraft** [comment]
Hard to tell..
but for sure non-one-handed guns shouldn't be allowed t be used with one handed ones.

Since they aren't one-handed it means they need **2** hands to use, so it would be unrealistic if someone would hold 2 weapons of different type at once.

**Comment by TypicalDarkness** [comment]
The problem is that we can't just remove the item in the offHand if the player scrolls over an incompatible item in the hotbar. (Where should we put it?)
We could add a property do define the accuracy if the gun is held in the offHand. (multiplier or absolute value? and what would the default be?).
As well as that we could increase the accuracy if there is no item held in the offHand.

Maybe we should only shoot the offHand gun if it has ammo/can reload and the mainHand holds an Item that has no left-click function (most things exept for swords and tools)

**Comment by TypicalDarkness** [comment]
A config would definitely be useful, but again, what should happen if the player has a gun in the off-hand and then scrolls to a non-oneHanded gun in the mainHand?

**Comment by Fexcraft** [comment]
I meant like.. since it already needs an "override" of the vanilla left-click,
shouldn't be a big deal to implement an check if the mainhand item is "valid"?
_(I guess I could figure something out if I would see the code, just guessing atm)_

**Comment by TypicalDarkness** [comment]
Preventing the gun from shooting is easy. I meant equipping the gun

**Comment by Fexcraft** [comment]
So if I get it right.. you mean to de-equip the gun in case the mainhand has an invalid item?

**Comment by TypicalDarkness** [comment]
Yes. We kind of have to do this because if the gun is held, the player expects that it can also shoot

**Comment by Fexcraft** [comment]
True that... _(humans >.<)_
How about checking if inventory has space and put into inventory,
and in case not, drop it. (ik this will annoy ppl, but it's the same in other games).

**Comment by TypicalDarkness** [comment]
Maybe it's possible to detect when an invalid gun is selected in the hotbar and then remove the offHand-stack, store it in a variable and put it back when the invalid gun is deselectd again.

**Comment by Fexcraft** [comment]
Or might be a bit hacky, but wouldn't need to move the gun anywhere from the offhand...
Render a different json model, like e.g. an red "X" when an invalid item is in mainhand.
_(maybe use some kind of metadata?)_

**Comment by TypicalDarkness** [comment]
Yes. that would be a good idea. We could use the barrier texture for example

**Comment by TypicalDarkness** [comment]
It actually works better than I expected and I didn't even have to use any hacky methods:
![2016-10-06_23 50 00](https://cloud.githubusercontent.com/assets/11584285/19171943/b34715d4-8c1f-11e6-9492-c41d5ed98ab6.png)

The problem is that I can only modify the model of guns this way, for vanilla items/items form other mods the cross would have to be rendered directly into the gui. Preventing the rendering of the non-gui model of this items (model that is "held"), while still displaying the unequip-animation right after the invalid item was selected, would be even harder.

**Comment by Fexcraft** [comment]
_mean solution: drop the offhand item_
